lib/galaxy/tools/parameters/basic.py

```diff
@@ -2268,14 +2268,14 @@ def to_param_dict_string( self, value, other_values={} ):
 
     # converts values to json representation:
     #   { id: LibraryDatasetDatasetAssociation.id, name: LibraryDatasetDatasetAssociation.name, src: 'lda' }
-    def to_json( self, value, app ):
+    def to_json( self, value, app, security ):
         if not isinstance( value, list ):
             value = [value]
         lst = []
         for item in value:
             encoded_id = encoded_name = None
             if isinstance(item, app.model.LibraryDatasetDatasetAssociation):
-                encoded_id = app.security.encode_id( item.id )
+                encoded_id = app.security.encode_id( item.id ) if security else item.id
```

Comment:
This var can be either encoded or not - we might want to rename it to prevent confusion.